src/main/java/handson/impl/CartService.java

```diff
@@ -52,4 +56,16 @@ public CartService(SphereClient client) {
         // TODO 5.1 Add discount code to cart
         return null;
     }
+    
+	/**
+	 * Add shipping address to the cart
+	 * 
+	 * @param cart
+	 *            the cart
+	 * @return completion state that sets the shipping address
+	 */
+	public CompletionStage<Cart> addShippingAddress(final Cart cart) {
```

Comment:
Thanks for addressing it! It might be more efficient (lees request) if you would define address already at cart creation step here: https://github.com/commercetools/commercetools-jvm-sdk-training/pull/15/files#diff-02f250fb579ea94ffffd398e37cee599L28

Reply:
yes that would be more efficient, but then i would have to give the answer to the exercise 3.1, which i think you want to leave open to students to answer. This is just a fix that prevents you from creating an order after u completed all the exercises because of the missing shipping address. See issue : https://github.com/commercetools/commercetools-jvm-sdk-training/issues/10

Reply:
True but this would also hide a fact that shipping address is mandatory for order creation as we would do it behind the scenes. It should be then part of the cart or order creation task. @nkuehn WDYT?